Library/Source/NativeEngine.cpp

```diff
@@ -222,6 +259,11 @@ namespace babylon
             // ALPHA_SCREENMODE: SRC + (1 - SRC) * DEST, SRC ALPHA + (1 - SRC ALPHA) * DEST ALPHA
             BGFX_STATE_BLEND_FUNC_SEPARATE(BGFX_STATE_BLEND_ONE, BGFX_STATE_BLEND_INV_SRC_COLOR, BGFX_STATE_BLEND_ONE, BGFX_STATE_BLEND_INV_SRC_ALPHA),
         };
+
+        constexpr std::array<bgfx::TextureFormat::Enum, 1> TEXTURE_FORMAT
+        {
+            bgfx::TextureFormat::RGBA32F
```

Comment:
rgba32f is 4 floats per pixel. Is it really the intended format? usually, it's rgba8

Reply:
I thought it was a bit weird too, but I chose this one because the format requested by Babylon.js was (in its own terms) `Engine.TEXTUREFORMAT_RGBA`, not `Engine.TEXTUREFORMAT_RGBA_INTEGER`, which was also an option. I may have misinterpreted what those options mean, though; I'll ask (here or offline) @bghgary or one of the other JavaScript-side experts to confirm which format should be used to back `Engine.TEXTUREFORMAT_RGBA`.

Reply:
Asked @deltakosh, turns out Babylon specifies whether a texture is byte or float in a different variable. I've put up another PR to Babylon.js to get that information properly sent to native, changes to native side coming up shortly.